moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -83,15 +112,73 @@ class PlanningComponent
     double max_velocity_scaling_factor;
     double max_acceleration_scaling_factor;
 
-    void load(const rclcpp::Node::SharedPtr& node)
+    void load(const rclcpp::Node::SharedPtr& node, const std::string& param_namespace = "")
     {
       std::string ns = "plan_request_params.";
-      node->get_parameter_or(ns + "planner_id", planner_id, std::string(""));
-      node->get_parameter_or(ns + "planning_pipeline", planning_pipeline, std::string(""));
-      node->get_parameter_or(ns + "planning_time", planning_time, 1.0);
-      node->get_parameter_or(ns + "planning_attempts", planning_attempts, 5);
-      node->get_parameter_or(ns + "max_velocity_scaling_factor", max_velocity_scaling_factor, 1.0);
-      node->get_parameter_or(ns + "max_acceleration_scaling_factor", max_acceleration_scaling_factor, 1.0);
+      if (!param_namespace.empty())
+      {
+        ns = param_namespace + ".plan_request_params.";
+      }
+      if (!node->get_parameter_or(ns + "planner_id", planner_id, std::string("")))
```

Comment:
Could this be simplified with [rosparam_shortcuts](https://github.com/PickNikRobotics/rosparam_shortcuts/tree/ros2)?

Reply:
As far as I know it is not used in the moveit2 codebase at this point and I have not used it yet. Uniform and good parameter handling is still and open issue. I've simplified the code for now by adding a template to remove the copy&paste code similar to e.g. moveit_servo